go/vt/vttablet/tabletserver/txthrottler/tx_throttler.go

```diff
@@ -286,44 +303,104 @@ func newTxThrottlerState(topoServer *topo.Server, config *txThrottlerConfig, tar
 		t.Close()
 		return nil, err
 	}
-	result := &txThrottlerState{
+	state := &txThrottlerState{
 		config:    config,
 		throttler: t,
 	}
-	createTxThrottlerHealthCheck(topoServer, config, result, target.Cell)
 
-	result.topologyWatchers = make(
-		[]TopologyWatcherInterface, 0, len(config.healthCheckCells))
-	for _, cell := range config.healthCheckCells {
-		result.topologyWatchers = append(
-			result.topologyWatchers,
+	// get cells from topo if none defined in tabletenv config
+	if len(config.healthCheckCells) == 0 {
+		ctx, cancel := context.WithTimeout(context.Background(), topo.RemoteOperationTimeout)
+		defer cancel()
+		var err error
+		if config.healthCheckCells, err = fetchKnownCells(ctx, topoServer); err != nil {
+			log.Errorf("txThrottler: failed to open throttler: %+v", err)
+			return nil, err
+		}
+		state.cellsFromTopo = true
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	state.stopHealthCheck = cancel
+	state.initHealthCheckStream(topoServer, target)
```

Comment:
nit: I think we should not update the `config.healthCheckCells` instead we should pass in `cells` to `initHealthCheckStream`

Reply:
@harshit-gangal that's a good point! Updating "config" on the fly seems wrong 👍 

We still need the `healthCheckCells` in a place where more than one thing can access it, though, because it's read in a few places but potentially updated by the `for` loop (a goroutine) in `healthChecksProcessorFactory`

The only cleaner option I can see is adding `healthCheckCells []string` to `txThrottlerState`, however this will require `newTxThrottlerState` to be passed a new `env tabletenv.Env` or `config tabletenv.TabletConfig`. What do you think?

Reply:
saw the last commit. seems reasonable to me.